core/src/edu/cornell/gdiac/main/NPCModel.java

```diff
@@ -0,0 +1,327 @@
+   * <p>
+   * The main purpose of this constructor is to set the initial capsule orientation.
+   */
+  public NPCModel() {
+    super(0, 0, 0.5f, 1.0f);
+    setFixedRotation(true);
+    isStop = false;
+    // Gameplay attributes
+    color = Color.WHITE;
+    setDensity(1);
+  }
+
+  /**
+   * Returns left/right movement of this character.
+   * <p>
+   * This is the result of input times player force.
+   *
+   * @return left/right movement of this character.
+   */
+  public float getMovement() {
+    return movement;
+  }
+
+  /**
+   * Sets left/right movement of this character.
+   * <p>
+   * This is the result of input times player force.
+   *
+   * @param value left/right movement of this character.
+   */
+  public void setMovement(float value) {
+    movement = value / 10F;
+    // Change facing if appropriate
+    if (movement < 0) {
+      faceRight = false;
+    } else if (movement > 0) {
+      faceRight = true;
+    }
+  }
+
+  public void setStop(boolean stop) {
+    isStop = stop;
+  }
+
+  /**
+   * Returns how much force to apply to get the player moving
+   * <p>
+   * Multiply this by the input to get the movement value.
+   *
+   * @return how much force to apply to get the player moving
+   */
+  public float getForce() {
+    return force;
+  }
+
+  /**
+   * Sets how much force to apply to get the player moving
+   * <p>
+   * Multiply this by the input to get the movement value.
+   *
+   * @param value how much force to apply to get the player moving
+   */
+  public void setForce(float value) {
+    force = value;
+  }
+
+  /**
+   * Returns how hard the brakes are applied to get a player to stop moving
+   *
+   * @return how hard the brakes are applied to get a player to stop moving
+   */
+  public float getDamping() {
+    return damping;
+  }
+
+  /**
+   * Sets how hard the brakes are applied to get a player to stop moving
+   *
+   * @param value how hard the brakes are applied to get a player to stop moving
+   */
+  public void setDamping(float value) {
+    damping = value;
+  }
+
+  /**
+   * Returns the upper limit on player left-right movement.
+   * <p>
+   * This does NOT apply to vertical movement.
+   *
+   * @return the upper limit on player left-right movement.
+   */
+  public float getMaxSpeed() {
+    return maxspeed;
+  }
+
+  /**
+   * Sets the upper limit on player left-right movement.
+   * <p>
+   * This does NOT apply to vertical movement.
+   *
+   * @param value the upper limit on player left-right movement.
+   */
+  public void setMaxSpeed(float value) {
+    maxspeed = value;
+  }
+
+  /**
+   * Initializes the player via the given JSON value
+   * <p>
+   * The JSON value has been parsed and is part of a bigger level file.  However, this JSON value is
+   * limited to the player subtree
+   *
+   * @param directory the asset manager
+   * @param json      the JSON subtree defining the player
+   */
+  public void initialize(AssetDirectory directory, JsonValue json, int gSizeY) {
```

Comment:
Just so you know you should not copy and paste large chunks of code like this. Walker is centralizing the JSON parsing logic so for now I'll let it slide but in the future try to create an abstraction for things like this.

Reply:
Thanks for letting me know. I just thought it would be easier if I could adapt from something similar that is already working. I will try to avoid that in the future!